script.js

```diff
@@ -0,0 +1,621 @@
+// only allow player to submit bid points at start of the round
+let canSubmitBidPoints = true;
+// Only allow player to click on dealButton after submitting bid points and before exchanging cards
+let canDealStartingCards = false;
+// only allow player to click on card after having dealt cards
+let canClickCards = false;
+// only allow player to click on exchangeOrHoldCardsButton after having dealt cards
+let canExchangeOrHoldCards = false;
+
+// Helper functions ================================================
+// create elements needed when browser loads
+const createStartingElements = () => {
+  // container to display the player's cardContainers
+  playerHandContainer = document.createElement('div');
+  playerHandContainer.setAttribute('id', 'player-hand-container');
+  playerHandContainer.classList.add('row', 'align-items-center', 'justify-content-center');
+
+  // container to display output messages informing player about the state of the game
+  gameInfo = document.createElement('div');
+  gameInfo.setAttribute('id', 'game-info-container');
+
+  // container to display buttons
+  buttonsContainer = document.createElement('div');
+  buttonsContainer.setAttribute('id', 'buttons-container');
+  buttonsContainer.classList.add('row');
+  // button to deal cards
+  dealButton = document.createElement('button');
+  dealButton.setAttribute('id', 'deal-button');
+  dealButton.classList.add('col-5', 'col-md-4');
+  // button to exchange cards
+  exchangeOrHoldCardsButton = document.createElement('button');
+  exchangeOrHoldCardsButton.setAttribute('id', 'exchange-button');
+  exchangeOrHoldCardsButton.classList.add('col-5', 'col-md-4');
+
+  // element to display heading for player's total points
+  totalPointsHeadingEl = document.createElement('p');
+  // element to display player's total points
+  totalPointsInfoEl = document.createElement('p');
+  // container to display player's total points info and heading
+  totalPointsContainer = document.createElement('div');
+  totalPointsContainer.setAttribute('id', 'total-points-container');
+  // element to display heading for player's bid points
+  bidPointsHeadingEl = document.createElement('p');
+  // element to display player's bid points
+  bidPointsInfoEl = document.createElement('p');
+  // input element to ask for points player wants to bid
+  bidPointsInputEl = document.createElement('input');
+  // button to submit the points player wants to bid
+  bidPointsButton = document.createElement('button');
+  // container to display player's bid points info and heading
+  bidPointsContainer = document.createElement('div');
+  bidPointsContainer.setAttribute('id', 'bid-points-container');
+  // container to display player's bid points input box and button
+  bidPointsInputContainer = document.createElement('div');
+  bidPointsInputContainer.setAttribute('id', 'bid-points-input-container');
+};
+
+// For creating a shuffled deck ------------------------------------
```

Comment:
Consider splitting the DOM functions into a separate file from the game logic functions

Reply:
By DOM functions do you mean creating the elements: `bidPointsInputContainer = document.createElement('div');` , setting the attribute and class names: `bidPointsInputContainer.setAttribute('id', 'bid-points-input-container');` ? 

If yes, then should I also declare the variables storing the DOM elements: `let bidPointsInputContainer;` in that same file? Asking this because I think my code will have errors when I run it if I do not declare the variable names in the same file. 
Let's say I place the JS file (containing the DOM functions I mentioned) before the JS file (containing the declarations and other code that call the DOM elements (eg. `bidPointsInputContainer.innerText`)), then because the variables have not been declared, the JS file (containing the DOM functions I mentioned) will have errors. If I place the JS file (containing the DOM functions I mentioned) after the JS file (containing the declarations and other code that call the DOM elements), then the other code that call the DOM elements (eg. `bidPointsInputContainer.innerText`) will have errors because it is read before `bidPointsInputContainer = document.createElement('div');`.

Thanks for reading my lengthy question!

Reply:
Hope we clarified during code review! Please let us know if we can help clarify further!